finality-provider/service/fp_instance.go

```diff
@@ -173,41 +171,42 @@ func (fp *FinalityProviderInstance) finalitySigSubmissionLoop() {
 	defer fp.wg.Done()
 
 	for {
+		// start submission in the first iteration
+		pollerBlocks := fp.getAllBlocksFromChan()
```

Comment:
nice, I bet we do this a lot throughout the codebases, that we wait before the first iteration

Reply:
I just realized that we can't do it for every loop for cases where we have `continue` upon error. If we do, then we the loop will go forever. Need to figure out other ways for it